### ISSUE TITLE
Migrate diagnostic codes from FE to BE

### ISSUE THREAD

**Issue by ajspotts** [department-of-veterans-affairs/caseflow#14872]
Blocked by #14871 

## Description
After creating a new table and model for our diagnostic codes, we need to ensure that all of our FE code is properly utilizing them as our new source of information.  The ask of this ticket is to identify all of our FE code that currently utilizes the JSON object where we had been storing the codes and update it to now receive their information from the new model.  Finally, we need to remove the JSON object as it will be obsolete after these changes

## Acceptance criteria
- [ ] Identify all of the places on the FE where we use diagnostic codes
- [ ] Update the code to now receive the codes from the BE
- [ ] Remove the giant diagnostic codes JSON [object](https://github.com/department-of-veterans-affairs/caseflow/blob/master/client/constants/DIAGNOSTIC_CODE_DESCRIPTIONS.json)

## Background/context/resources
[originating](https://github.com/department-of-veterans-affairs/caseflow/issues/10948) issue
[slack](https://dsva.slack.com/archives/C54QCEHAL/p1595355732017200) convo detailing issue discovery
[slack](https://dsva.slack.com/archives/CQCUMFA0H/p1595431749007500) convo detailing decision process

**Comment by hschallhorn** [comment]
```bash
git grep DIAGNOSTIC_CODE_DESCRIPTIONS
app/models/decision_issue.rb:    validates :diagnostic_code, inclusion: { in: Constants::DIAGNOSTIC_CODE_DESCRIPTIONS.keys.map(&:to_s) },
app/models/decision_issue.rb:    if diagnostic_code && Constants::DIAGNOSTIC_CODE_DESCRIPTIONS[diagnostic_code]
app/models/decision_issue.rb:      description = Constants::DIAGNOSTIC_CODE_DESCRIPTIONS[diagnostic_code]["status_description"]
app/models/issue.rb:      diagnostic_code_description = Constants::DIAGNOSTIC_CODE_DESCRIPTIONS[diagnostic_code]
app/models/request_issue.rb:    if diagnostic_code && Constants::DIAGNOSTIC_CODE_DESCRIPTIONS[diagnostic_code]
app/models/request_issue.rb:      description = Constants::DIAGNOSTIC_CODE_DESCRIPTIONS[diagnostic_code]["status_description"]
client/app/queue/AddEditIssueView.jsx:import DIAGNOSTIC_CODE_DESCRIPTIONS from '../../constants/DIAGNOSTIC_CODE_DESCRIPTIONS';
client/app/queue/AddEditIssueView.jsx:  renderDiagnosticCodes = () => _.keys(DIAGNOSTIC_CODE_DESCRIPTIONS).map((value) => ({
client/app/queue/SelectDispositionsView.jsx:import DIAGNOSTIC_CODE_DESCRIPTIONS from '../../constants/DIAGNOSTIC_CODE_DESCRIPTIONS';
client/app/queue/SelectDispositionsView.jsx:          options={_.map(Object.keys(DIAGNOSTIC_CODE_DESCRIPTIONS), (key) => ({ label: key,
client/app/queue/mtv/checkout/AddEditDecisionIssueModal.jsx:import DIAGNOSTIC_CODE_DESCRIPTIONS from '../../../../constants/DIAGNOSTIC_CODE_DESCRIPTIONS';
client/app/queue/mtv/checkout/AddEditDecisionIssueModal.jsx:        options={Object.keys(DIAGNOSTIC_CODE_DESCRIPTIONS).map((key) => ({
client/app/queue/utils.js:import DIAGNOSTIC_CODE_DESCRIPTIONS from '../../constants/DIAGNOSTIC_CODE_DESCRIPTIONS';
client/app/queue/utils.js:  const readableLabel = DIAGNOSTIC_CODE_DESCRIPTIONS[code];
spec/factories/vacols/case_issue.rb:    Constants::DIAGNOSTIC_CODE_DESCRIPTIONS.each_key do |diag_code|
spec/factories/vacols/case_issue.rb:      trait_name = Constants::DIAGNOSTIC_CODE_DESCRIPTIONS[diag_code]["status_description"]
```

**Comment by hschallhorn** [comment]
_[what is this chart?](https://github.com/department-of-veterans-affairs/caseflow/issues/11533)_
```bash
1 | 
2 | 
3 | ||||
5 | ||||||||
8 | 
```

Pass data in the initial react app?

Keep an eye on performance; file follow-up ticket to improve.
Scope this ticket to have no visible user changes.

Include diagnostic description whenever diagnostic code is returned to the FE.

Consider using the user-searcher, only populate on typing structure. Might not be good if you users want to be able to browser.

Create a new route to get all diagnostic codes and description for displaying a dropdown list for users to select.

#### Why 3?
* Won't need much extra to grab the description of a specific code, pass to the front end where needed

#### Why 5?
* lots to consider performance-wise
* New endpoint to grab all diagnostic codes on page load
